src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3616,20 +3616,18 @@ export class ResultsViewPageStore {
 
     readonly germlineConsentedSamples = remoteData<SampleIdentifier[]>(
         {
-            await: () => [this.studyIds, this.samples],
+            await: () => [this.studyIds, this.sampleMap],
             invoke: async () => {
-                const germlineConsentedSamples = await fetchGermlineConsentedSamples(
+                const germlineConsentedSamples: SampleIdentifier[] = await fetchGermlineConsentedSamples(
                     this.studyIds,
                     AppConfig.serverConfig.studiesWithGermlineConsentedSamples
                 );
-                const sampleIds = this.samples.result
-                    ? this.samples.result.map(s => s.sampleId)
-                    : [];
 
                 // do not simply return all germline consented samples,
                 // only include the ones matching current sample selection
+                const sampleMap = this.sampleMap.result!;
                 return germlineConsentedSamples.filter(s =>
-                    sampleIds.includes(s.sampleId)
+                    sampleMap.has(s, ['sampleId', 'studyId'])
```

Comment:
@adamabeshouse  don't samples have unique id? in which case we could halve this lookup time?   i realize this would be a small optimization compared to what you just did

Reply:
`germlineConsentedSamples` dont have the unique id attached to them, but this lookup time is still just one level of indexing